mesa/space.py

```diff
@@ -256,29 +256,34 @@ def get_neighborhood(
         if neighborhood is not None:
             return neighborhood
 
-        coordinates: set[Coordinate] = set()
-
         x, y = pos
-        for dy in range(-radius, radius + 1):
-            for dx in range(-radius, radius + 1):
-                # Skip coordinates that are outside manhattan distance
-                if not moore and abs(dx) + abs(dy) > radius:
-                    continue
-
-                coord = (x + dx, y + dy)
 
-                if self.out_of_bounds(coord):
-                    # Skip if not a torus and new coords out of bounds.
-                    if not self.torus:
+        range_x = self._compute_inbound_ranges(x, self.width, radius)
+        range_y = self._compute_inbound_ranges(y, self.height, radius)
+
+        neighborhood = []
+
+        for nx in range_x:
+            for ny in range_y:
+                if not moore:
+                    dx, dy = abs(nx - x), abs(ny - y)
+                    if self.torus:
+                        # In the torus case the deviation from the given point
+                        # can derive from traversing the grid from the outside
+                        if (dx_out := self.width - dx) < dx:
+                            dx = dx_out
+                        if (dy_out := self.height - dy) < dy:
+                            dy = dy_out
+                    # Skip coordinates that are outside manhattan distance
+                    if dx + dy > radius:
                         continue
-                    coord = self.torus_adj(coord)
 
-                coordinates.add(coord)
+                coord = (nx, ny)
+                neighborhood.append(coord)
 
-        if not include_center:
-            coordinates.discard(pos)
+        if not include_center and neighborhood:
```

Comment:
I think another constraint to consider when designing the code is that whether it would be easier to translate to Numba/Cython. It would be harder with `set`.

Reply:
yes clearly true, lists are easier to pass to those libraries and also the speed up boost is much more relevant

Reply:
what do you think about my last comment on points outside the grid?